### PR TITLE
Update documentation for comms-deduplication v2

### PR DESCRIPTION
**Pull request by l-catallo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,20 +1,34 @@
...
 
 It is based on two main concepts:
 
-- `id`: The unique identifier of the signal
-- `processorId`: The unique identifier of the system that processes the signal
+- `id`: An identifier of the operation, this can be any value that is both
+  consistent and unique for each operation. _Only one[*](#limitations)_
+  operation with the same `id` will be allowed to run.
+- `contextId`: A context identifier, usually the type of operation being
+  performed. This is used to keep differnt types of operation separate within a
```
Comment (rcdmrl) [inline README.md:18]:
```suggestion
  performed. This is used to keep different types of operation separate within a
```

File: README.md [hunk 2]
```diff
@@ -29,35 +43,165 @@ Then add this snippet to your `build.sbt`
...
+  dedup <- MeteorDeduplication.resource[IO, String, String](
+    client,
+    dedupTable,
+    dedupConf
+  )
+} yield dedup
+
+dedupResource.use { deduplication =>
+
+  // Create two contexts to deduplicate the operations separately
```
Comment (rcdmrl) [inline README.md:123]: Good example!

**Review by rcdmrl — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -29,35 +43,165 @@ Then add this snippet to your `build.sbt`
...
+  val sendEmailCtx = deduplication.context[String]("sendEmail")
+  val storeEmailCtx = deduplication.context[Unit]("storeEmail")
+
+  events
+    .evalMap { evt =>
+      for {
+        // Wrap the operations in a protect call
+        sendId <- sendEmailCtx.protect(evt.id, sendEmail(evt))
```
Comment (rcdmrl) [inline README.md:131]: I think that if I just jumped into this example I could have the following question:
> what happens when `sendEmailCtx.protect(evt.id, sendEmail(evt)` detects a duplicate? Does it fail the `F` thus leading to short-circuiting behaviour?

Comment (l-catallo) [reply README.md:132]: Good shout 👍

**Pull request merged**